CeeFM/Decideify
# Suggestions Algorithm Work Part 2

**Description**

This pull request continues the arduous work of creating the specific suggestion algorithm for each content type, as all of the responses and the data they return will vary slightly (to immensely) and need to fit inside of the same class/object in our application. We have pivoted from working in the TVShows.js to actually working on saving to our database from the Books.js component.

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/Decideify)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-suggestions` as one of the available branches
- `git checkout cm-suggestions` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-suggestions`
- `Start Decideify.sln` to open the project in Visual Studio
- If you haven’t created the Decideify database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Seed_Users_DB.sql)

- Click the execute with debugger solid green button to launch the app
- You should be directed to the /login page and prompted to login - enter “[test@gmail.com](mailto:test@gmail.com)” in the email field, enter "thistbh" as the password and click “Login” - this should redirect you back to the home page.
- Go to the "Books" content hub.
- Click the "Test The Book API" button. Nothing should appear to happen, but rest assured a fetch call is being made to the NYT Books API.
- Make sure your DevTools menu is open and directed to the console. Click the "Print Books" button, and you should see an array of objects appear with hundreds of book results across the NYT's various bestsellers lists. Below that, you should also see a single object that just contains data about one book - specifically Sarah J. Maas's "House of Flame and Shadow".
- Click the "Save Book" button and information about Sarah J. Maas's "House of Flame and Shadow" will be saved to the Suggestion table of the database with your currently logged in user's id appended, along with other data (a mixture of real data and placeholder data as I sort out the remaining kinks in the database).
- Run this SQL command to double-check if your entry has been saved
```
SELECT * FROM Suggestion
```

Checklist
